lib/view_component/codemods/v3_slot_setters.rb

```diff
@@ -159,9 +191,16 @@ def view_component_path_glob
         "{#{view_component_paths.join(",")}}"
       end
 
+      def rails_view_paths
+        ActionController::Base.view_paths.select do |path|
+          path.to_s.include?(Rails.root.to_s)
+        end.map(&:to_s)
+      end
+
       def view_paths
         @view_paths ||= [
-          "app/views",
+          rails_view_paths,
+          Rails.application.config.view_component.preview_paths,
```

Comment:
Does this break when running the codemod in a gem that does not require Rails?

Reply:
In gems app rake tasks require dummy app and at least development rails dependency:
```
# Rakefile
APP_RAKEFILE = File.expand_path("test/dummy/Rakefile", __dir__)
load "rails/tasks/engine.rake"
```
And can be executed with the prefix:
```
bin/rails app:view_component:detect_legacy_slots
```
Considering that, I don't think Rails view paths call will be a problem.